mvlearn/embed/kcca.py

```diff
@@ -111,53 +167,48 @@ def fit(self, Xs):
         """
         Xs = check_Xs(Xs, multiview=True)
 
-        components_ = kcca(
-            Xs,
-            self.reg,
-            self.n_components,
-            ktype=self.ktype,
-            sigma=self.sigma,
-            degree=self.degree,
-        )
+        x = Xs[0]
+        y = Xs[1]
 
-        self.weights_ = _listdot(Xs, components_)
+        N = len(x)
+        x = _center_norm(x)
+        y = _center_norm(y)
 
-        return self.weights_
+        self.Kx = _make_kernel(x, self.ktype, self.constant,
+                               self.degree, self.sigma)
+        self.Ky = _make_kernel(y, self.ktype, self.constant,
+                               self.degree, self.sigma)
 
-    def transform(self, Xs):
-        """
-        Uses KCCA weights to transform Xs into canonical components
-        and calculates correlations.
+        Id = np.eye(N)
+        Z = np.zeros((N, N))
+        dim = min(x.shape[1], y.shape[1])
 
-        Parameters
-        ----------
-        vdata: float
-               Standardized data (z-score)
+        # Solving eigenvalue problem
+        R = 0.5*np.r_[np.c_[self.Kx, self.Ky], np.c_[self.Kx, self.Ky]]
+        D = np.r_[np.c_[self.Kx+self.reg*Id, Z], np.c_[Z, self.Ky+self.reg*Id]]
 
-        Returns
-        -------
-        weights_ : list of array-likes
-                   Canonical weights
-        preds_: list of array-likes
-                Prediction components of test dataset
-        corrs_: list of array-likes
-                Correlations on the test dataset
-        """
+        betas = linalg.eig(R, D)[0]  # eigenvalues
+        alphas = linalg.eig(R, D)[1]  # right eigenvectors
+        ind = np.argsort(np.sum(np.diag(betas), axis=0), axis=0)
 
-        if not hasattr(self, "weights_"):
-            raise NameError("kCCA has not been trained.")
+        perm_mat = np.zeros((len(ind), len(ind)))
+
+        for idx, i in enumerate(ind):
+            perm_mat[idx, i] = 1
+
+        alphass = np.real(np.dot(alphas, perm_mat)/np.linalg.norm(alphas))
 
-        self.components_ = _listdot([d.T for d in Xs], self.weights_)
-        self.cancorrs_ = _listcorr(self.components_)
-        self.cancorrs_ = self.cancorrs_[np.nonzero(self.cancorrs_)]
+        # weights
+        weight1 = np.asarray(alphass[:N, :dim])
+        weight2 = np.asarray(alphass[N:, :dim])
+        self.weights_ = [weight1, weight2]
 
         return self
 
-    def fit_transform(self, Xs):
+    def transform(self, Xs):
```

Comment:
addressed your first paragraph.

not sure what you mean by computing the kernel(Xs_transform, Xs_fit). to create the kernel for the transformed data, you do not need the x_fit (i tested it to make sure and it messed things up).

i removed the self.Kx and self.Ky and made it just Kx and Ky to save space. the original Kx and Ky are only used to find the weights. afterwards, and for the transform, it can recompute it if the Xs_transform is the same as Xs_fit. i think adding a parameter will make things more complicated than they need to be.